planetiler-core/src/main/java/com/onthegomap/planetiler/util/LoopLineMerger.java

```diff
@@ -75,16 +77,48 @@ private void merge() {
       if (node.getEdges().size() == 2) {
         Edge a = node.getEdges().getFirst();
         Edge b = node.getEdges().get(1);
+        mergeTwoEdges(a, b);
         node.getEdges().clear();
-        List<Coordinate> coordinates = new ArrayList<>();
-        coordinates.addAll(a.coordinates.reversed());
-        coordinates.addAll(b.coordinates.subList(1, b.coordinates.size()));
-        Edge c = new Edge(a.to, b.to, coordinates, a.length + b.length);
-        a.to.removeEdge(a.reversed);
-        b.to.removeEdge(b.reversed);
-        a.to.addEdge(c);
-        if (a.to != b.to) {
-          b.to.addEdge(c.reversed);
+      }
+    }
+  }
+
+  private void mergeTwoEdges(Edge a, Edge b) {
+    List<Coordinate> coordinates = new ArrayList<>();
```

Comment:
This assumes a.to = b.from, we could also put `assert a.to == b.from;` here to check that is true when the jvm is run with `-ea` (enable assertions) argument, which is always set when running unit tests.

Reply:
Good idea